### PR TITLE
Clear user when an account is successfully closed (marked for deletion)

### PR DESCRIPTION
**Pull request by andrewserong**
This PR fixes an issue where if a user creates an account but doesn't complete site signup, then navigates to `/` and then decides to delete their account at `/me/account/close`, their signup state still remains. If they navigate to `/start` they will be stuck in the same signup flow they were in before, and unable to reach the `user` step. The only way to reset signup is to clear their cache, or log in and log out with another user account. If the user proceeds to try to complete signup, they will get stuck.

The fix proposes calling `user.clear()` when a request to close an account is successful, similar to what we do when we log out a user.

#### Before

After deleting an account and navigating back to `/`, this is what happens when a user clicks on Get Started:

delete-user-before

If they proceed to try to complete signup they'll run into a bunch of errors:

image

#### After

Here's a more complete demonstration of the issue resolved, going from creating a user account, navigating to delete the user account, and then navigating back to `/start`:

delete-user

#### Changes proposed in this Pull Request

* Call `user.clear()` to clear out the current user and signup state when an account is closed (marked for deletion)

#### Testing instructions

* Open an incognito window
* Go to `/start` and create a new user account but don't complete signup
* Navigate to `/me/account/close`
* In the console, look at `state.account` and the `isClosed` property should be `false`
* Delete this newly created account
* In the console, look at `state.account` and the `isClosed` property should be `true`
* Navigate to `/start`
* You should now be on the `user` step of signup
* In the console, look at `state.account` and the `isClosed` property should be `false`
* Create another new account and complete signup as normal, and confirm that your site is created

Fixes https://github.com/Automattic/zelda-private/issues/259

### CODE REVIEW

**Review by ramonjd — APPROVED** [review]
Works as described. Thanks @andrewserong !

Also clearing the user store resolves https://github.com/Automattic/zelda-private/issues/259

**Comment by ramonjd** [comment]
I think this is good to go, but I might hold off on merging until there are folks around (in case of revert : ) )

**Comment by lancewillett** [comment]
Would this close #28196 also, by any chance? Happy to test after this lands.

**Comment by andrewserong** [comment]
@lancewillett I'm just going to rebase this PR and then merge it in. Unfortunately, I don't think this one will fix that other bug, because this only adds the behaviour to clear out the user state when an account is closed, rather than when the user is halfway through the signup flow loading screen. I was wondering if https://github.com/Automattic/wp-calypso/pull/38330 might have fixed it, but I don't think so. Manually testing, it looks like reloading the loading screen page restores the user to the first step in the signup flow for that issue, so I think it'd be good to fix, but it doesn't seem critical to me. I think the best bet would be to prioritise this with @Automattic/zelda — I've just added #28196 to the project board.

**Comment by andrewserong** [comment]
Looks like the IE canaries are failing consistently. I think it could be to do with changes in the domains/suggestions endpoint: p1578019048015000-slack-e2e-testing-discuss — so, not related to the change in this PR, but since it's a failing canary, to be safe I'll hold off on merging for now.

**Pull request merged**
